components/kf-monitoring/MetricsExporter.go

```diff
@@ -0,0 +1,161 @@
+package monitoring_util
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/log"
+	"net"
+	"net/http"
+)
+
+// Common label keys for all metrics signals
+// COMPONENT: name of component outputing the metrics, eg. "tf-operator"
+const COMPONENT = "component"
+// KIND: each componenet can label their metrics with custom tag "kind". Suggest keeping "kind" value to be CONSTANT PER METRICS.
+const KIND = "kind"
+const NAMESPACE = "namespace"
+// ACTION: request action type of the metrics, eg. "CRUD"
+const ACTION = "action"
+// SEVERITY: level of importance, used to filter alerts
+const SEVERITY  = "severity"
+
+const SEVERITY_MINOR = "Minor"
+const SEVERITY_MAJOR = "Major"
+const SEVERITY_CRITICAL = "Critical"
+
+const METRICSPORT = 8079
+const METRICSPATH = "/metrics"
+
+var (
+	// Counter metrics
+	// num of requests counter vec
+	requestCounter = prometheus.NewCounterVec(
```

Comment:
Nit: It would be more readable, if these vars are only defined in `var (...)`with zero-values, but assigned with values in `init()` right before it is used.